### DISCUSSION TITLE
When hiding or displaying the spectrogram, there will be changes in the waveform.  Why does this happen?  Are there any solutions to this issue?

### DISCUSSION THREAD

**Discussion by Wei-forward** [katspaugh/wavesurfer.js#3460] [Q&A]
## Bug description
When hiding or displaying the spectrogram, there will be changes in the waveform.  Why does this happen?  Are there any solutions to this issue?

## Environment
 - Browser: Chrome

## Minimal code snippet
let element = this.wavesurfer.getWrapper();
      let spectrogramBox = element.querySelector(".cursor").nextElementSibling;

      if (value) {
        spectrogramBox.style.visibility = "visible";
        element.style.height = "428px";
      } else {
        spectrogramBox.style.visibility = "hidden";
        element.style.height = "228px";
      }

## Expected result
Waveform invariant

**Comment by katspaugh** [comment]
Looks like it’s triggering a resize observer when you change the height of the container.

`getWrapper` should be used only by plugins, try to avoid manipulating wavesurfer elements.

Hiding/showing the spectrogram can be done via destroying and re-registering the plugin.

**Comment by Wei-forward** [comment]
I registered the spectrogram using the registerPlugin, but the page didn't show up

**Comment by katspaugh** [comment]
This is now (hopefully) fixed in 7.6.1. Please let me know if it fixed it for you.
